src/runtimes/node/bundlers/nft/es_modules.ts

```diff
@@ -148,21 +149,21 @@ const shouldTranspile = (
 
 const transpileESM = async ({
   basePath,
+  cache,
   config,
   esmPaths,
-  fsCache,
   reasons,
   name,
 }: {
   basePath: string | undefined
+  cache: RuntimeCache
   config: FunctionConfig
   esmPaths: Set<string>
-  fsCache: FsCache
   reasons: NodeFileTraceReasons
   name: string
 }) => {
-  const cache: Map<string, boolean> = new Map()
-  const pathsToTranspile = [...esmPaths].filter((path) => shouldTranspile(path, cache, esmPaths, reasons))
+  const shouldCompileCache: Map<string, boolean> = new Map()
```

Comment:
I feel that this comment would help contextualise this variable in relation to the `cache` argument.

```suggestion
  // Used for memoizing the check for whether a path should be transpiled.
  const shouldCompileCache: Map<string, boolean> = new Map()
```